content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,163 @@
+---
+title: Save
+---
+
+Save patterns help users to store and update their content and configuration throughout GitHub. These changes should be represented in the UI accurately, quickly, and obviously. Their behavior should inspire confidence and trust.
+
+## Calls to action
+
+### Label copy
+
+All configuration flows should have calls to action that include an obvious active verb such as "Create", "Save", "Delete", or "Update" (for example, "Create" a new repository, "Add" an SSH key to your profile, "Save" security preferences, "Update" a repository's description, "Delete" an old email address).
+
+When defining these calls to actions, make sure to:
+1. Keep the text as succinct and clear as possible
+2. Add the item’s name to the button text in cases where it may be unclear to the user what is being changed
+
+![Screenshot of the 'Create a new repository' form](https://user-images.githubusercontent.com/2313998/150868902-a8514e84-5cd4-4081-b13d-46f00f404625.png)
+
+### Placement
+
+Save buttons should be placed somewhere intuitive and easily accessible. If all of the fields save automatically, do not include a save button. This makes it unclear when data is being saved.
+
+Default to placing the submit button at the bottom right of the form. In cases where this is an unintuitive placement, explore other positions to place the button.
+<img width="1790" alt="Screenshot of the 'Edit repository details' form" src="https://user-images.githubusercontent.com/2313998/150868909-71f72f1a-eafc-4777-a639-96d81b33eebb.png">
+
+{/*
+TODO:
+Consider how we might define more patterns to keep the "Save" button easily accessible on long forms:
+- add save and cancel buttons on top AND bottom?
+- use a sticky header with save and cancel buttons?
+*/}
+
+If a save button is used to save an inline edit to the input or inputs where the change was made.
+![Screen recording of a user editing a PR's title](https://user-images.githubusercontent.com/2313998/150868914-6af36067-39a6-4fec-adb9-518c5abd595d.gif)
+
+It's strongly recommended to have a single save button for all of the inputs on the page. If you're designing a page with pieces of content that can be separately toggled into an edit mode (such as editing an issue title), only allow one edit mode toggled on at a time so that only one save button is visible. When there is on save button per section of a page or form, users may be confused about whether the save button will save all of the inputs or just the subset of the inputs. This confusion is even more likely for visually impaired users because they can't see if each save button is visually grouped with a set of inputs.
+
+{/* 
+TODO: find an example of this, mock one up (if that's allowed), or exclude this visual example
```

Comment:
Any ideas for _good_ examples of a form with subsection that gets submitted with a "Save" button, and a separate subsection that automatically saves?

The closest thing I could find was the Codespaces settings - you have to explicitly click "save" for one section, but the rest automatically save.

Reply:
Ignore this comment. I'm going to be replacing all screenshots with mockups that demonstrate the issue in isolation.